doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -595,90 +584,85 @@ the conversion in hypotheses :n:`{+ @ident}`.
 
-   + They unfold a constant if and only if it leads to a :math:`\iota`-reduction,
+   + It unfolds constants only if they lead to an :math:`\iota`-reduction,
      i.e. reducing a match or unfolding a fixpoint.
-   + While reducing a constant unfolding to (co)fixpoints, the tactics
-     use the name of the constant the (co)fixpoint comes from instead of
+   + When reducing a constant unfolding to (co)fixpoints, the tactic
+     uses the name of the constant the (co)fixpoint comes from instead of
      the (co)fixpoint definition in recursive calls.
 
-   The :tacn:`cbn` tactic was intended to be a more principled, faster and more
-   predictable replacement for :tacn:`simpl`.
-
-   The :tacn:`cbn` tactic accepts the same flags as :tacn:`cbv` and
-   :tacn:`lazy`. The behavior of both :tacn:`simpl` and :tacn:`cbn`
-   can be tuned using the :cmd:`Arguments` command.
+   :n:`@occurrences`
+     Permits selecting whether to reduce the conclusion and/or one or more
+     hypotheses.  The `at` option of :n:`@occurrences` is not allowed here,
+     but :n:`reference_occs` and :n:`@pattern_occs` have an somewhat less
+     flexible `at` option for selecting specific occurrences.
 
-   .. todo add "See <subsection about controlling the behavior of reduction strategies>"
-      to TBA section
+   Only transparent constants whose name can be reused in
+   recursive calls may be unfolded by :tacn:`simpl`. For instance, a
+   constant :g:`plus' := plus` may be unfolded and reused in
+   recursive calls, but a constant such as :g:`succ := plus (S O)` is
+   never unfolded.
 
-   Notice that only transparent constants whose name can be reused in the
-   recursive calls are possibly unfolded by :tacn:`simpl`. For instance a
-   constant defined by :g:`plus' := plus` is possibly unfolded and reused in
-   the recursive calls, but a constant such as :g:`succ := plus (S O)` is
-   never unfolded. This is the main difference between :tacn:`simpl` and :tacn:`cbn`.
-   The tactic :tacn:`cbn` reduces whenever it will be able to reuse it or not:
-   :g:`succ t` is reduced to :g:`S t`.
+   :n:`{| @reference_occs | @pattern_occs }` can limit the application of :tacn:`simpl`
+   to:
 
-.. tacv:: cbn [ {+ @qualid} ]
-          cbn - [ {+ @qualid} ]
+   - the applicative subterms whose head occurrence is the unfoldable
+     constant :n:`@qualid` or is the constant used
+     in the notation :n:`@string` (see :n:`@reference`)
+   - subterms matching a pattern :n:`@one_term`
 
-   These are respectively synonyms of :n:`cbn beta delta [ {+ @qualid} ] iota zeta`
-   and :n:`cbn beta delta - [ {+ @qualid} ] iota zeta` (see :tacn:`cbn`).
+   The behavior can be tuned using the :cmd:`Arguments` command.
 
-.. tacv:: simpl @pattern
+.. tacn:: cbn {? @reductions } {? @occurrences }
 
-   This applies :tacn:`simpl` only to the subterms matching
-   :n:`@pattern` in the current goal.
+   :tacn:`cbn` tactic was intended to be a more principled, faster and more
```

Comment:
```suggestion
   The :tacn:`cbn` tactic was intended to be a more principled, faster and more
```
or
```suggestion
   :tacn:`cbn` was intended to be a more principled, faster and more
```

Reply:
"was" or "is"?  "was" suggests that it may not have achieved that goal.  IIUC it's not used nearly as often as `simpl`.

Reply:
>"was" suggests that it may not have achieved that goal. 

That's the correct interpretation. Right now both tactics have edge cases and no one can be said to be uniformly better.